en/docs/chapter_searching/replace_linear_by_hashing.md

```diff
@@ -1,14 +1,14 @@
 # Hash optimization strategies
 
-In algorithm problems, **we often reduce the time complexity of algorithms by replacing linear search with hash search**. Let's use an algorithm problem to deepen understanding.
+In algorithm problems, **we often reduce an algorithm's time complexity by replacing a linear search with a hash-based search**. Let's use an algorithm problem to deepen the understanding.
```

Comment:
reduce an algorithm's time complexity => reduce the time complexity of an algorithm